geomesa-kafka-bundle/geomesa-kafka-processors/src/main/scala/org/geomesa/nifi/processors/kafka/GeoMesaKafkaProcessor.scala

```diff
@@ -27,13 +29,8 @@ abstract class GeoMesaKafkaProcessor extends AbstractGeoIngestProcessor(GeoMesaK
 }
 
 object GeoMesaKafkaProcessor extends PropertyDescriptorUtils {
-
-  import KafkaDataStoreFactoryParams._
-
   // note: KafkaDataStoreFactory.ParameterInfo is consumer-oriented, but we want producer properties here
-  private val KafkaProperties = {
-    val params =
-      Seq(Brokers, Zookeepers, ZkPath, ProducerConfig, SerializationType, TopicPartitions, TopicReplication)
-    params.map(createPropertyDescriptor)
-  }
+  private val KafkaProperties =
+    createPropertyDescriptors(KafkaDataStoreFactory) ++
+        Seq(ProducerConfig, TopicPartitions, TopicReplication).map(createPropertyDescriptor)
```

Comment:
Ok, I'm a little confused on the need to do this for ProducerConfig, TopicPartitions, and TopicReplication.  @elahrvivaz can you explain that whenever you get around to reading through GitHub messages again?

Reply:
it's b/c for geoserver we only return the 'read' related props, i.e. https://github.com/locationtech/geomesa/blob/main/geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaDataStoreFactory.scala#L75
so the default descriptors that get returned don't include the producer props. maybe we should update the factory so that all the props are returned, and they can just be ignored in geoserver...

Reply:
Ok, this is looking good enough for right now.  If you want to make some changes like that later, I'm all for it.  It seems like we are in a position where NiFi and GeoServer will now see the sensible configs for a producer and a consumer, right?

Reply:
yeah I think so, consumer for geoserver and producer for nifi